src/arrayvec.rs

```diff
@@ -101,12 +101,43 @@ macro_rules! array_vec {
 /// assert_eq!(no_ints.len(), 0);
 /// ```
 #[repr(C)]
-#[derive(Clone, Copy)]
 pub struct ArrayVec<A> {
   len: u16,
   pub(crate) data: A,
 }
 
+impl<A> Clone for ArrayVec<A>
+where
+  A: Array + Clone,
+  A::Item: Clone,
+{
+  #[inline]
+  fn clone(&self) -> Self {
+    Self { data: self.data.clone(), len: self.len }
+  }
+
+  #[inline]
+  fn clone_from(&mut self, o: &Self) {
+    let iter = self
+      .data
+      .as_slice_mut()
+      .iter_mut()
+      .zip(o.data.as_slice())
+      .take(self.len.max(o.len) as usize);
+    for (dst, src) in iter {
+      dst.clone_from(src)
+    }
+    self.len = o.len;
```

Comment:
Might want to take-and-drop all the elements from `o.len..self.len`.

Reply:
Just remember to skip this part if `o.len >= self.len`

Reply:
If I follow you right, we'd want to have something like:

if o.len < self.len {
  self[o.len..].iter_mut().for_each(|r| take(r));
}

immediately after the for loop?

Reply:
```rust
if let Some(to_drop) = self.data.get_mut(o.len() .. self.len()) {
  to_drop.for_each(|x| take(x));
}
```